src/app/portlets/content-types/create-edit/content-types-form/content-types-form.component.ts

```diff
@@ -52,117 +55,146 @@ export class ContentTypesForm extends BaseComponent {
         name: '',
         owner: '',
         system: false,
-        variable: '',
         workflow: ''
     };
     private expireDateFieldOptions: SelectItem[] = [];
     private publishDateFieldOptions: SelectItem[] = [];
-    private sitesOrFolderOptions: Observable<SelectItem[]>;
+    private sitesOrFolderOptions = [];
     private workflowOptions: SelectItem[] = [];
+    private icon: string;
+    private type: string;
 
     constructor(messageService: MessageService, private siteService: SiteService,
         private crudService: CrudService, private loginService: LoginService,
         private stringUtils: StringUtils, private renderer: Renderer,
-        private fb: FormBuilder, private router: Router) {
+        private fb: FormBuilder, private router: Router, private route: ActivatedRoute,
+        private contentTypesInfoService: ContentTypesInfoService) {
         super([
-            'Variable',
-            'description',
-            'Identifier',
-            'Publish-Date-Field',
+            'Detail-Page',
             'Expire-Date-Field',
             'Host-Folder',
-            'Detail-Page',
-            'Workflow',
+            'Identifier',
+            'Properties',
+            'Publish-Date-Field',
+            'URL-Map-Pattern-hint1',
             'URL-Pattern',
+            'Variable',
+            'Workflow',
             'cancel',
-            'save',
-            'message.contentlet.required',
+            'description',
             'fields',
-            'Properties',
-            'URL-Map-Pattern-hint1'
+            'message.contentlet.required',
+            'save',
+            'update'
         ], messageService);
     }
 
     ngOnInit(): void {
-        this.initFormGroup();
-        this.initFormData();
         this.initHostFieldOptions();
         this.initWorkflowtFieldOptions();
         this.initDatesFieldOptions();
+        this.actionButtonLabel = this.i18nMessages['save'];
+        let url = this.route.snapshot.url[0].path;
+
+        if (url === 'create') {
```

Comment:
I like to handle this such as 
let urlHandler = new Array ();

urlHandler ['create'] = function() { this.setIcon ....}
urlHandler ['edit'] = function() { this.setEditMode ....}

urlHandler[url]();

But it is just an idea to avoid if, else, else pattern :)